pkgs/tools/networking/boundary/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, lib, fetchurl, unzip }:
+let
+  version = "0.1.4";
+
+  system = stdenv.hostPlatform.system;
+  suffix = {
+    x86_64-linux = "linux_amd64";
+    aarch64-linux = "linux_arm64";
+    x86_64-darwin = "darwin_amd64";
+  }."${system}" or (throw "Unsupported system: ${system}");
+
+  fetchsrc = sha256: fetchurl {
+    url = "https://releases.hashicorp.com/boundary/${version}/boundary_${version}_${suffix}.zip";
+    sha256 = sha256."${system}";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "boundary";
+  inherit version;
+
+  src = fetchsrc {
+    x86_64-linux = "14f4mc5l04ammszvcq7dv8a6pvdz13k5y6r226lqs72lfyxi5kc7";
+    aarch64-linux = "1blzn42birpq2pbqz63ffzd0p2xcgrgmg2q6jmmabfjzmp0z7ax2";
+    x86_64-darwin = "03w1c3p45nz1h3cf705d3n15c4rxsf1r448lcqkxaarsicxs32gw";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
```

Comment:
```suggestion
```
Please disable phases you do not want to run.

Reply:
`doCheck = false;` and `doBuild = false;` don't actually do anything
This was the only method I saw in the shell script https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh


`doCheck = false; doBuild = false; doPatch = false;`:

```
λ nix-build -A boundary
this derivation will be built:
  /nix/store/6v4gci5dvb8557gsqh8ws8md76i4yf7g-boundary-0.1.4.drv
building '/nix/store/6v4gci5dvb8557gsqh8ws8md76i4yf7g-boundary-0.1.4.drv'...
unpacking sources
Archive:  /nix/store/nxn230ymbl3xpi2vb0bblpbjia5yk490-boundary_0.1.4_linux_amd64.zip
  inflating: boundary
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/6fnk449a7hg7454jihi3hlqpada4v0b9-boundary-0.1.4
shrinking /nix/store/6fnk449a7hg7454jihi3hlqpada4v0b9-boundary-0.1.4/bin/boundary
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
strip is /nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/6fnk449a7hg7454jihi3hlqpada4v0b9-boundary-0.1.4/bin
patching script interpreter paths in /nix/store/6fnk449a7hg7454jihi3hlqpada4v0b9-boundary-0.1.4
checking for references to /build/ in /nix/store/6fnk449a7hg7454jihi3hlqpada4v0b9-boundary-0.1.4...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/6fnk449a7hg7454jihi3hlqpada4v0b9-boundary-0.1.4
```

`phases = [ "unpackPhase" "installPhase" ];`:

```
λ nix-build -A boundary
this derivation will be built:
  /nix/store/km1p7pg2sfvl7pd5i3qmwq0658hhmh20-boundary-0.1.4.drv
building '/nix/store/km1p7pg2sfvl7pd5i3qmwq0658hhmh20-boundary-0.1.4.drv'...
unpacking sources
Archive:  /nix/store/nxn230ymbl3xpi2vb0bblpbjia5yk490-boundary_0.1.4_linux_amd64.zip
  inflating: boundary
installing
/nix/store/p9nc9cxcc3if5cc5cr5kxhvqf9mqvrvn-boundary-0.1.4
```

Reply:
stdenv is a bit special. Normal packages should not overwrite phases because this can break hooks in unexpected ways.

Is the binary linked statically? Can you disable that while building? Last resort would be to disable striping with ``doStrip = false;`` and a comment why it is required.

Reply:
I'm not building boundary so I can't disable it being statically linked while building

`doStrip = false;` doesn't do anything either:

```
# Without doStrip = false;
λ nix-build -A boundary
this derivation will be built:
  /nix/store/7x9xshjknn7w72qf9lkm74n7xi2z98na-boundary-0.1.4.drv
building '/nix/store/7x9xshjknn7w72qf9lkm74n7xi2z98na-boundary-0.1.4.drv'...
unpacking sources
Archive:  /nix/store/nxn230ymbl3xpi2vb0bblpbjia5yk490-boundary_0.1.4_linux_amd64.zip
  inflating: boundary
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/3ya4kcivdim0gn9sb4f627l57wrlnzya-boundary-0.1.4
shrinking /nix/store/3ya4kcivdim0gn9sb4f627l57wrlnzya-boundary-0.1.4/bin/boundary
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
strip is /nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/3ya4kcivdim0gn9sb4f627l57wrlnzya-boundary-0.1.4/bin
patching script interpreter paths in /nix/store/3ya4kcivdim0gn9sb4f627l57wrlnzya-boundary-0.1.4
checking for references to /build/ in /nix/store/3ya4kcivdim0gn9sb4f627l57wrlnzya-boundary-0.1.4...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/3ya4kcivdim0gn9sb4f627l57wrlnzya-boundary-0.1.4

# With doStrip = false;
λ nix-build -A boundary
this derivation will be built:
  /nix/store/11aa6rfri0f09pknh186973h6c0b526g-boundary-0.1.4.drv
building '/nix/store/11aa6rfri0f09pknh186973h6c0b526g-boundary-0.1.4.drv'...
unpacking sources
Archive:  /nix/store/nxn230ymbl3xpi2vb0bblpbjia5yk490-boundary_0.1.4_linux_amd64.zip
  inflating: boundary
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/l7rl2b5nd2i3vwvkpgzi8n8vjzp0s092-boundary-0.1.4
shrinking /nix/store/l7rl2b5nd2i3vwvkpgzi8n8vjzp0s092-boundary-0.1.4/bin/boundary
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
strip is /nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/l7rl2b5nd2i3vwvkpgzi8n8vjzp0s092-boundary-0.1.4/bin
patching script interpreter paths in /nix/store/l7rl2b5nd2i3vwvkpgzi8n8vjzp0s092-boundary-0.1.4
checking for references to /build/ in /nix/store/l7rl2b5nd2i3vwvkpgzi8n8vjzp0s092-boundary-0.1.4...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/l7rl2b5nd2i3vwvkpgzi8n8vjzp0s092-boundary-0.1.4
```

Just leaving it still packages it fine but it just outputs a bunch of useless messages & warnings

Reply:
It is ``dontStrip = true;``. I always mix them up.

Reply:
`dontStrip = true;` doesn't remove any of the patchelf outputs and the stripping wasn't a problem
Also with stripping is quite a bit smaller than without

With `dontStrip = true;`
```
building '/nix/store/akj27gsbaidixgn0183620pg77aav7qi-boundary-0.1.4.drv'...
unpacking sources
Archive:  /nix/store/nxn230ymbl3xpi2vb0bblpbjia5yk490-boundary_0.1.4_linux_amd64.zip
  inflating: boundary
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/qjg7swlcr9xym1kdfshb6bfhp5kks0xj-boundary-0.1.4
shrinking /nix/store/qjg7swlcr9xym1kdfshb6bfhp5kks0xj-boundary-0.1.4/bin/boundary
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
patching script interpreter paths in /nix/store/qjg7swlcr9xym1kdfshb6bfhp5kks0xj-boundary-0.1.4
checking for references to /build/ in /nix/store/qjg7swlcr9xym1kdfshb6bfhp5kks0xj-boundary-0.1.4...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/qjg7swlcr9xym1kdfshb6bfhp5kks0xj-boundary-0.1.4
λ ls -al result/bin/
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  .
.r-xr-xr-x root root  47 MB Thu Jan  1 01:00:01 1970  boundary
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  ..
```

Without `dontStrip = true;`
```
λ nix-build -A boundary
this derivation will be built:
  /nix/store/xzc2f8pl8zwp4cfa2f1p4mjscdwm7ial-boundary-0.1.4.drv
building '/nix/store/xzc2f8pl8zwp4cfa2f1p4mjscdwm7ial-boundary-0.1.4.drv'...
unpacking sources
Archive:  /nix/store/nxn230ymbl3xpi2vb0bblpbjia5yk490-boundary_0.1.4_linux_amd64.zip
  inflating: boundary
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/q1im5968j0bzqnbpnr740xjxjfqcpr6i-boundary-0.1.4
shrinking /nix/store/q1im5968j0bzqnbpnr740xjxjfqcpr6i-boundary-0.1.4/bin/boundary
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/q1im5968j0bzqnbpnr740xjxjfqcpr6i-boundary-0.1.4/bin
patching script interpreter paths in /nix/store/q1im5968j0bzqnbpnr740xjxjfqcpr6i-boundary-0.1.4
checking for references to /build/ in /nix/store/q1im5968j0bzqnbpnr740xjxjfqcpr6i-boundary-0.1.4...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/q1im5968j0bzqnbpnr740xjxjfqcpr6i-boundary-0.1.4
λ ls -al result/bin/
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  .
.r-xr-xr-x root root  39 MB Thu Jan  1 01:00:01 1970  boundary
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  ..
```

---

Other `dontX` options


`dontPatchELF = true;` does remove one of the two `patchelf` lines

```
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/mfw66a963hx3y8gk2k1ncp6yjgfhpm5r-boundary-0.1.4/bin
patching script interpreter paths in /nix/store/mfw66a963hx3y8gk2k1ncp6yjgfhpm5r-boundary-0.1.4
checking for references to /build/ in /nix/store/mfw66a963hx3y8gk2k1ncp6yjgfhpm5r-boundary-0.1.4...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/mfw66a963hx3y8gk2k1ncp6yjgfhpm5r-boundary-0.1.4
```

`dontPatchShebangs = true;` removes the `patching script interpreter paths` line

```
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/0gm3r0h3a4nqfxfdz81nj6bb01b7nz4f-boundary-0.1.4/bin
checking for references to /build/ in /nix/store/0gm3r0h3a4nqfxfdz81nj6bb01b7nz4f-boundary-0.1.4...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/0gm3r0h3a4nqfxfdz81nj6bb01b7nz4f-boundary-0.1.4
```

Also `dontConfigure = true;` and `dontBuild = true;` worked